src/cpp/src/sampler.cpp

```diff
@@ -755,7 +755,8 @@ SamplerOutput Sampler::sample(std::vector<SequenceGroup::Ptr> & sequence_groups,
             continue;
 
         size_t num_running_sequences = sequence_group->num_running_seqs();
-        size_t actual_seq_len = sequence_group->get_num_scheduled_tokens(); // points to a token which needs to be sampled
+        size_t actual_seq_len = sequence_group->is_matmul_sliced() ?
+            sequence_group->get_seq_len_to_sample() : sequence_group->get_num_scheduled_tokens(); // points to a token which needs to be sampled
         size_t padded_amount_of_processed_tokens = std::max(actual_seq_len, batch_seq_len);
```

Comment:
@olpipi could you please drop `padded_amount_of_processed_tokens` and use `actual_seq_len` instead on other places? 
Padding is obsolete feature and now dropped from CB.